src/driver/database-connection.ts

```diff
@@ -15,8 +15,8 @@ export interface DatabaseConnection {
 
 export interface QueryResult<O> {
   /**
-   * This is defined for update and delete queries and contains
-   * the number of rows the query updated/deleted.
+   * This is defined for insert, update and delete queries and contains
+   * the number of rows the query inserted/updated/deleted.
    */
   readonly numUpdatedOrDeletedRows?: bigint
```

Comment:
You're right, this should now be renamed to `numAffectedRows` now. I sucks that this will break 3rd party dialects, but we could give a heads up for the biggest one (SST and data api). 

We could even go so far as to handle the old property `numUpdatedOrDeletedRows` internally using ugly `any` casts. Then the old dialects would _mostly_ work but the maintainers would get type errors about the missing `numUpdatedOrDeletedRows` when they update their dialects. What do you think? Or is it better to just break it and expedite the update? We'll bump the version to indicate a breaking change of course.

Reply:
We could even open a PR [here](https://github.com/serverless-stack/kysely-data-api)

Reply:
If we detect the usage of `numUpdatedOrDeletedRows` we could have a one-time warning log indicating that the dialects is outdated?

Reply:
> We could even go so far as to handle the old property numUpdatedOrDeletedRows internally using ugly any casts. Then the old dialects would mostly work but the maintainers would get type errors about the missing numUpdatedOrDeletedRows when they update their dialects. What do you think? Or is it better to just break it and expedite the update? We'll bump the version to indicate a breaking change of course.

Our heaviest dependant is SST. Kysely is a hard dependency @^0.21.3 in their core package, and a peer in data-api @0.x.
Another potential heavy dependant is kysely-planetscale (100+ stars, used by an influencer with a big community). Kysely is a peer @*.

We could break things in the next major, or deprecate+warn and postpone break to major after next. Both are fine.
Either way, I'd go the extra mile and immediately open a pull request @ kysely-data-api and kysely-planetscale.